generic/tests/unattended_install.py

```diff
@@ -11,4 +18,58 @@ def run(test, params, env):
     :param params: Dictionary with the test parameters.
     :param env: Dictionary with test environment.
     """
-    unattended_install.run(test, params, env)
+    # use vm name to differentiate among single/multivm
+    vms = params.get("vms").split()
+    nr_vms = len(vms)
+    multivm = nr_vms > 1
+
+    if not multivm:
+        unattended_install.run(test, params, env)
+        return
+
+    def thread_func(vm):
+        """
+        Thread Method to trigger the unattended installation
+
+        :param vm: VM name
+        """
+        global error_flag
+        try:
+            vm_params = params.object_params(vm)
+            vm_params["main_vm"] = vm
+            unattended_install.run(test, vm_params, env)
+        except Exception as info:
+            logging.error(info)
+            error_flag = True
+
+    trigger_time = int(params.get("install_trigger_time", 0))
+    random_trigger = params.get("random_trigger", "no") == "yes"
+    thread_timeout = float(params.get("install_timeout", 200))
+    install_threads = []
+
+    if not params.get("master_images_clone"):
+        test.cancel("provide the param `master_images_clone` to clone"
+                    "images for vms")
+
+    for vm in vms:
+        thread = threading.Thread(target=thread_func, args=(vm,))
+        install_threads.append(thread)
+
+    for thread in install_threads:
+        if trigger_time:
+            sleep_time = trigger_time
+            if random_trigger:
+                sleep_time = random.randint(0, trigger_time)
+            time.sleep(sleep_time)
+            # increase thread_timeout based on the sleep time so that
+            # last VM could get actual thread_timeout
+            thread_timeout += sleep_time
```

Comment:
@luckyh I have taken care to address the `thread_timeout` in this way,
let `thread_timeout` initialize with `install_timeout`, if we trigger vm installation with different trigger time then cumulate it with `thread_timeout` so that last VM would get actual `thread_timeout`. If all the VMs are triggered at same time simultaneously then every VM will get the same `install_timeout` as `thread_timeout`.